modular_skyrat/modules/customization/controllers/subsystem/alt_titles.dm

```diff
@@ -0,0 +1,35 @@
+//SKYRAT MODULE CUSTOMIZATION
+// Make a seperate subsystem so it doesn't conflict with the actual job one.
+
+SUBSYSTEM_DEF(altjob)
+	name = "Alt Jobs"
+	init_order = INIT_ORDER_JOBS
+	flags = SS_NO_FIRE
+
+/datum/controller/subsystem/altjob/Initialize(timeofday)
+	RegisterSignal(SSdcs, COMSIG_GLOB_JOB_AFTER_SPAWN, .proc/on_job_after_spawn)
+	return ..(timeofday)
+
+/datum/controller/subsystem/altjob/proc/on_job_after_spawn(datum/source, datum/job/job, mob/living/spawned, client/player_client)
+	SIGNAL_HANDLER
+
+	var/alt_title = player_client.prefs.alt_job_titles[job.title]
+	var/mob/living/carbon/human/human = spawned
+
+	// If alt title is null, the player is just using default.
+	// Player also needs to be a human or we're aborting.
+	if(alt_title && istype(human))
+		// ID card
+		var/obj/item/card/id/card = human.get_idcard()
+		if(istype(card))
+			card.assignment = alt_title
+			card.update_label()
+
+		// PDA
+		for(var/obj/item/pda/pda in human)
```

Comment:
Is there no better way to do this?

Reply:
I'm trying to remember now why I didn't do `H.get_item_by_slot(pda_slot)`. It might have been out of a worry that jobs, quirks, etc have the theoretical ability to scramble things around. Would it be a good compromise to just say if there's an edge case where the PDA was remove out of the PDA slot that the person will need to update their PDA info?

Likewise with wallets I should have just used the slot because that actually is enforced (I think). Not sure why I did that.

So in general I did the subsystem to try to modularize it out of endless job modifications, which the `customization` module already does (this is part of why I put it in that module, so customization can sort of just own taking over the job datum). Would you rather I modify `/datum/outfit/job/post_equip()` and the wallet trait?

Reply:
Due to the fact we're already struggling with super long round starts, yeah, I think this could be a good move, it should reduce the impact of all of this on performance, hopefully.